### ISSUE TITLE
Create Training Loop

### ISSUE THREAD

**Issue by stuartfong1** [stuartfong1/improving-badly-drawn-bunnies#12]
Create the main for loop that trains the model. This involves getting a batch from the dataloader, encoding the sketches, sampling from the distribution, decoding the latent vector, calculating the loss, and then updating the model weights.
